sheerwater_benchmarking/utils/caching.py

```diff
@@ -423,30 +555,72 @@ def write_individual_raster(driver, bucket, ds, cache_key):
             write_individual_raster(driver, bucket, ds, cache_key)
 
 
-def cache_exists(backend, cache_path, verify_path=None):
-    """Check if a cache exists generically."""
+def cache_exists(backend, cache_path, verify_path=None, cache_local=False):
+    """Check if a cache exists, with local cache support.
+
+    Args:
+        backend (str): The backend to check
+        cache_path (str): The path to the cache (either cache path or null path)
+        verify_path (str): The path to the verify file
+        cache_local (bool): Whether to check the local cache
+
+    Returns:
+        tuple: A tuple containing a boolean indicating if the cache exists and
+            appropriate cache path
+    """
+    # If local caching is enabled, check the local cache first
+    fs = fsspec.core.url_to_fs(cache_path, storage_options=CACHE_STORAGE_OPTIONS)[0]
+    if cache_local and backend not in ['postgres', 'terracotta', 'delta']:
+        local_cache_path = get_modified_cache_path(cache_path, modification='local')
+        local_verify_path = get_modified_cache_path(verify_path, modification='local')
+        local_fs = fsspec.core.url_to_fs(local_cache_path, storage_options=LOCAL_CACHE_STORAGE_OPTIONS)
+        # If the local cache exists, AND the remote cache exists (not corrupted), return True
+        if _cache_exists(backend, local_cache_path, local_verify_path) and \
+                _cache_exists(backend, cache_path, verify_path):
+            if backend in ['zarr', 'parquet', 'pickle']:
+                # If we have a local cache, check and make sure the verify timestamp is the same
+                local_verify_ts = local_fs.open(local_verify_path, 'r').read()
+                remote_verify_ts = fs.open(verify_path, 'r').read()
+                if local_verify_ts == remote_verify_ts:
+                    return True
+            else:
+                return True
+        # If the remote cache exists, copy the remote cache to the local cache
+        if _cache_exists(backend, cache_path, verify_path):
+            fs.get(cache_path, local_cache_path, recursive=True)
+            if verify_path:
+                fs.get(verify_path, local_verify_path, recursive=True)
+            return True
+        else:
+            return False
```

Comment:
Should we throw a warning for cache_local with a backend that doesn't support it?